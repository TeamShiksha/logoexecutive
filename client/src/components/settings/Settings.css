.sett-cont-heading {
	color: var(--smooth-purple);
	margin-bottom: 32px;
	text-align: left;
	font-size: 32px;
}
.settings {
	display: flex;
	flex-direction: column;
	justify-content: center;
	padding: 64px 32px;
}

.settings-heading {
	margin: 0px 0px 20px;
}

.setting-instruction {
	font-size: var(--medium-text);
	margin: 6px 0px 30px;
	color: var(--description);
}

.settings-button {
	background-color: var(--smooth-purple);
	color: var(--white);
	border: none;
	border-radius: var(--bradius);
	cursor: pointer;
	padding: 12px;
	font-size: var(--medium-text);
	font-weight: 500;
	width: fit-content;
}
.settings-button:disabled {
	background-color: var(--faded-purple);
}
.settings-delete {
	background-color: rgba(255, 0, 0, 0.7);
}

.settings-confirm-message {
	margin-top: 10px;
}

.settings-success-bg {
	height: 100%;
	width: 100%;
	background: linear-gradient(90deg, rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.3));
	z-index: 1;
	-webkit-backdrop-filter: blur(2px);
	backdrop-filter: blur(2px);
	position: fixed;
	inset: 0;
	display: flex;
	justify-content: center;
	align-items: center;
}

.delete-error-msg {
	color: var(--error);
}

@media screen and (max-width: 768px) {
	.settings {
		padding: 32px 16px;
	}
}
