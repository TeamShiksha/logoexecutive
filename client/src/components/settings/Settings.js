import {useContext, useEffect, useState} from 'react';
import Modal from '../common/modal/Modal';
import {useApi} from '../../hooks/useApi';
import {useNavigate} from 'react-router-dom';
import {AuthContext} from '../../contexts/AuthContext';
import useCountdownTimer from '../../hooks/useCountdownTimer';
import ResponseCard from '../common/responseCard/ResponseCard';
import {FaCheck} from 'react-icons/fa6';
import './Settings.css';

function Settings() {
	const [modalOpen, setModalOpen] = useState(false);
	const [showDeleteResponseModal, setShowDeleteResponseModal] = useState(false);
	const [countdown, setCountdown] = useState(3);
	const navigate = useNavigate();
	const {logout} = useContext(AuthContext);
	const {data, errorMsg, loading, makeRequest, isSuccess} = useApi({
		url: 'api/user/delete',
		method: 'delete',
	});

	const openModal = () => setModalOpen(true);

	const handleConfirm = async () => {
		try {
			const response = await makeRequest();
			if (response) {
				setModalOpen(false);
				setTimeout(() => {
					logout();
				}, 3000);
			}
		} catch (error) {
			console.error('Failed to delete account: ', error);
		}
	};

	useEffect(() => {
		if (errorMsg) {
			setShowDeleteResponseModal(true);
			setModalOpen(false);
		} else {
			setShowDeleteResponseModal(false);
		}
	}, [errorMsg]);

	useCountdownTimer(isSuccess, navigate, countdown, setCountdown);

	return (
		<section className='settings' data-testid='testid-settings'>
			<h1 className='sett-cont-heading'>Settings</h1>
			<button className='settings-button'>Download Account Data</button>
			<p className='setting-instruction'>
				This will prepare a PDF download of your personal data.
			</p>
			<button onClick={openModal} className='settings-button settings-delete'>
				Delete Account
			</button>
			<p className='setting-instruction'>
				This will permanently delete your account and all associated data.
			</p>
			<Modal
				modalOpen={modalOpen}
				setModal={setModalOpen}
				showButtons={true}
				handleConfirm={handleConfirm}
				loading={loading}
			>
				<h2>Are you sure?</h2>
				<p className='settings-confirm-message'>
					Please confirm that you want to permanently delete your account. This
					action cannot be undone, and you will lose all your data and access to
					our services.
				</p>
			</Modal>
			{isSuccess && (
				<div className='settings-success-bg'>
					<ResponseCard
						countdown={countdown}
						message={data?.message}
						Icon={<FaCheck className='response-success-icon' />}
						title={'Delete Account Successful'}
						redirectTo='home page'
					/>
				</div>
			)}
			{showDeleteResponseModal && (
				<Modal
					modalOpen={showDeleteResponseModal}
					showButtons={false}
					showCloseIcon={true}
					setModal={setShowDeleteResponseModal}
				>
					<p className='delete-error-msg'>
						Something went wrong. Please try after some time.
					</p>
				</Modal>
			)}
		</section>
	);
}

export default Settings;
