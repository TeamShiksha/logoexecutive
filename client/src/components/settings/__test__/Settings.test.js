import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import Settings from '../Settings';

describe('Settings component', () => {
	test('renders Settings component correctly', () => {
		/** Arrange */
		const {asFragment} = render(<Settings />);
		/** Assert */
		// Snapshot testing for the initial rendering
		expect(asFragment()).toMatchSnapshot();
	});

	test('opens modal when "Delete Account" button is clicked', () => {
		/** Arrange */
		render(<Settings />);

		/** Act */
		// Click the "Delete Account" button
		fireEvent.click(screen.getByText('Delete Account'));

		/** Assert */
		// Now, the modal should be open
		expect(screen.getByText('Are you sure?')).toBeInTheDocument();
	});
});
