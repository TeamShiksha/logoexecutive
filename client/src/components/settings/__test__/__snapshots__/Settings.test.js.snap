// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders Settings component correctly 1`] = `
<DocumentFragment>
  <section
    class="settings"
  >
    <h1
      class="sett-cont-heading"
    >
      Settings
    </h1>
    <button
      class="settings-button"
    >
      Download Account Data
    </button>
    <p
      class="setting-instruction"
    >
      This will prepare a PDF download of your personal data.
    </p>
    <button
      class="settings-button settings-delete"
    >
      Delete Account
    </button>
    <p
      class="setting-instruction"
    >
      This will permanently delete your account and all associated data.
    </p>
  </section>
</DocumentFragment>
`;
